//python -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. alicia.proto

syntax = "proto3";

service EditDistanceService {
    rpc EditDistance(EditDistanceRequest) returns (EditDistanceResponse) {}
}

message EditDistanceRequest {
    int32 process_id = 1;
    int32 leader = 2;
    string first_word = 3;
    string second_word = 4;
}
  
message EditDistanceResponse {
    int32 distance = 1;
}

service RingElection {
    rpc StartElection(Message) returns (Message) {}
}

message Message {
    int32 origin = 1;
    int32 max_id = 2;
    int32 rounds = 3;
    int32 leader = 4;
}

service BerkeleySynchronization {
    rpc RequestTime(Empty) returns (TimeResponse) {}
    rpc AdjustTime(TimeAdjustment) returns (Empty) {}
}

message TimeResponse {
    int64 time = 1;
}

message TimeAdjustment {
    int64 adjustment = 1;
}

message Empty {}
